File: solutions/julia/triangle/1/triangle.jl
function is_equilateral(sides)
    if sides[1] == 0
        return false
    end
    if sides[1]==sides[2] && sides[2]==sides[3]
        return true
    end
    return false
end

function is_isosceles(sides)
    if (sides[1]+sides[2]) < sides[3] ||
      (sides[1]+sides[3]) < sides[2] ||
      (sides[2]+sides[3]) < sides[1]
        return false
    end
    if sides[1]==sides[2] || sides[1]==sides[3] ||
        sides[2] == sides[3]
        return true
    end
    return false
end

function is_scalene(sides)
    if (sides[1]+sides[2])< sides[3] ||
        (sides[1]+sides[3])< sides[2] ||
        (sides[2]+sides[3]) < sides[1]
          return false
      end
      if sides[1]==sides[2] || sides[1]==sides[3] ||
        sides[2] == sides[3]
        return false
    end
    return true
end

File: solutions/csharp/triangle/1/Triangle.cs
using System;

public static class Triangle
{
    public static bool IsScalene(double side1, double side2, double side3)
    {
        if (side1 > side2+side3 || side2 > side1+side3 || side3 > side1+side2) return false;
        if (!IsEquilateral(side1, side2, side3) &&
            !IsIsosceles(side1, side2, side3)) return true;
            return false;
        //throw new NotImplementedException("You need to implement this function.");
    }

    public static bool IsIsosceles(double side1, double side2, double side3)
    {   if (side1 > side2+side3 || side2 > side1+side3 || side3 > side1+side2) return false;
        if (side1 == side2 || side1 == side3 || side2 == side3) return true;
        return false;
        //throw new NotImplementedException("You need to implement this function.");
    }

    public static bool IsEquilateral(double side1, double side2, double side3)
    {
        if (side1 == 0) return false;
        if (side1 == side2 && side2 == side3) return true;
        return false;
        //throw new NotImplementedException("You need to implement this function.");
    }
}